blogs/cli/index.md

```diff
@@ -0,0 +1,444 @@
+
+#### Encrypted traffic
+
+What about encrypted traffic ? 
+
+Most of the time, your traffic will be secured and you will not have access to the data it holds. However, Wireshark has the ability to decrypt the content if you [provide the proper secrets](https://wiki.wireshark.org/TLS#tls-decryption). However, getting these secrets may be tricky depending on where the encryption happens and needs some juggling. 
+If you are looking for the traffic between your browser and the cluster, you can simply run the following command to capture [pre-master-secret keylog file](https://wiki.wireshark.org/TLS#using-the-pre-master-secret):
+```
+SSLKEYLOGFILE="keylog.txt" chromium-browser &
+```
+
+This generated file contains each client and server traffic secrets to decode your packets.
+
+Then in Wireshark, specify the file path in `Edit→Preferences→Protocols→TLS→(Pre)-Master-Secret log filename`. A single file can hold keys for multiple TLS connections.
+
+![wireshark tls keylog](./images/wireshark_tls_keylog.png)
+
+You should now have access to decrypted packets !
+
+What about a secure connection between two pods ? 
+For these cases, you need to have a good understanding of what's involved behind the scene. If your application rely on OpenSSL, GnuTLS or NSS TLS library, you should be able to set the `SSLKEYLOGFILE` enviroment variable and extract the result.
+
+For our nodejs example, you need to either use the cli option [--tls-keylog](https://nodejs.org/api/cli.html#cli_tls_keylog_file) or modify the code and subscribe on [keylog event](https://nodejs.org/api/tls.html#event-keylog). When multiple teams are involved, that may take a while !
+
+Another alternative is to get a man in the middle approach using a proxy. This can be done using [kubetap](https://soluble-ai.github.io/kubetap/).
+
+You can install it using krew by running:
+```
+oc krew install tap
+```
+
+Then you can run the following command to port forward the traffic of your service to the proxy:
+```
+oc tap on -n packet-capture-scenario -p 3001 --https nodejs-basic --port-forward
+```
+
+The output should look like this:
+```
+Establishing port-forward tunnels to Service...
+                                                                                     
+
+Port-Forwards:
+
+  mitmproxy - http://127.0.0.1:2244
+  nodejs-basic - http://127.0.0.1:4000
+```
+
+Open the mitm proxy url and generate a query to your route. You will see queries listed in the proxy web UI with the the content displayed:
+
+![mitm proxy](./images/mitm_proxy.png)
+
+## Future improvments
+
+For now, the Network Observability CLI is only providing `capture commands` such as [many other observability projects](https://github.com/collabnix/kubetools?tab=readme-ov-file#observability) but it will be improve to embedd tools to facilitate your life as much as possible.
```

Comment:
```suggestion
For now, the Network Observability CLI is only providing `capture commands` such as [many other observability projects](https://github.com/collabnix/kubetools?tab=readme-ov-file#observability) but it will improve to embed tools to facilitate your life as much as possible.
```